web/settings/settings.go

```diff
@@ -64,20 +65,20 @@ func (h *HTTPHandler) getSessions(c echo.Context) error {
 	return jsonapi.DataList(c, http.StatusOK, objs, nil)
 }
 
-func (h *HTTPHandler) warnings(c echo.Context) error {
+func (h *HTTPHandler) listWarnings(c echo.Context) error {
 	inst := middlewares.GetInstance(c)
 
 	// Any request with a token can ask for the context (no permissions are required)
 	if _, err := middlewares.GetPermission(c); err != nil && !isMovedError(err) {
 		return err
 	}
 
-	warnings := inst.Warnings()
-	if warnings == nil {
-		warnings = []*jsonapi.Error{}
+	w := warnings.List(inst)
+	if w == nil {
+		w = []*jsonapi.Error{}
 	}
```

Comment:
[nit] I don't think we need to assign something to `w` when it is empty. These 3 lines may be removed.